docs/ref/models/expressions.txt

```diff
@@ -450,6 +450,137 @@ Conditional expressions allow you to use :keyword:`if` ... :keyword:`elif` ...
+
+    >>> from django.db.models import OuterRef, Subquery
+    >>> newest = Comment.objects.filter(post=OuterRef('pk')).order_by('-created_at')
+    >>> Post.objects.annotate(newest_commenter_email=Subquery(newest.values('email')[:1]))
+
+In this case, the every item in the Post queryset will be annotated with the
+email address of the author of the newest comment on that post.
+
+
+.. note::
+
+  The examples in this section are designed to show how to force
+  Django to execute a subquery. In some cases it may be possible to
+  write an equivalent queryset that performs the same task more
+  clearly or efficiently.
+
+Referencing columns from the outer queryset
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. class:: OuterRef(field)
+
+.. versionadded:: 1.11
+
+Used when a queryset that will be used within a ``Subquery`` needs to refer to
+a field from the outer query. It acts like an ``F`` expression, but the check
+to see if it refers to a valid field is not made until the outer queryset is
+resolved.
+
+Instances of ``OuterRef`` may be used in conjunction with nested instances
+of ``Subquery`` to refer to a containing queryset that is not the immediate
+parent::
+
+    >>> books = Book.objects.filter(author=OuterRef(OuterRef('pk')))
+
+This queryset would need to be within a nested pair of `Subquery` instances
+to resolve correctly.
+
+Limiting the subquery to a single column
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are times when a single column must be returned from a ``Subquery``. For
+instance, to use a ``Subquery`` as the target of an ``__in`` lookup::
+
+    >>> posts = Post.objects.filter(published_at__lte=now() - timedelta(1))
+    >>> Comment.objects.filter(post__in=Subquery(posts.values('pk')))
+
+In this case, the subquery must only return a single column: the primary
+key of the post.
+
+Limiting the subquery to a single row
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To prevent multiple rows being returned in a subquery, a slice (``[:1]``)
+of the queryset is used. Using a `.get()` will fail, because the ``OuterRef``
+can not be resolved until the queryset is used within a ``Subquery``::
+
+    >>> Post.objects.annotate(newest_commenter_email=Subquery(newest.values('email')[:1]))
+
+In this case, the subquery must only return a single column _and_ a single
+row: the email address of the most recently created comment.
+
+``Exists()`` subqueries
+~~~~~~~~~~~~~~~~~~~~~~~
+
+.. class:: Exists(queryset)
+
+.. versionadded:: 1.11
+
+``Exists`` is a ``Subquery`` subclass that uses an SQL ``EXISTS`` statement. In
+many cases it will perform better than a subquery since the database is able to
+stop evaluation of the subquery when the first matching row is found::
+
+    >>> from django.db.models import Exists, OuterRef
+    >>> recent_comments = Comment.objects.filter(post=OuterRef('pk'), created_at__gte=one_day_ago)
+    >>> Post.objects.filter(recent_comment=Exists(recent_comments))
```

Comment:
Oops, this doesn't work (and the next section says so -- `FieldError: Cannot resolve keyword 'recent_comment' into field.`). I revised it as I saw fit.